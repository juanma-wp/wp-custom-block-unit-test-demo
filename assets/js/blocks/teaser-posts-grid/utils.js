/**
 * Update an item in an array of items by index.
 *
 * @param {Array}  items   Original array.
 * @param {Object} newItem Updated item.
 * @param {Number} index   Index of item to update.
 */
export function updateItem( items, updatedItem, index ) {
	return [
		...items.slice( 0, index ),
		{
			...items[ index ],
			...updatedItem,
		},
		...items.slice( index + 1, items.length ),
	];
}

/**
 * Delete an item from an array of items.
 *
 * @param {Array}  items Original array.
 * @param {Number} index Index
 */
export function deleteItem( items, index ) {
	return [
		...items.slice( 0, index ),
		...items.slice( index + 1, items.length ),
	];
}

/**
 * Append a new item to the end of an array of items.
 *
 * @param {Array}  items   Original array.
 * @param {Object} newItem New item to append.
 */
export function appendItem( items, newItem ) {
	return [
		...items,
		newItem,
	]
}
