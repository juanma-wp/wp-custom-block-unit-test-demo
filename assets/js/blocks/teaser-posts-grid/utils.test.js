import { updateItem, deleteItem, appendItem } from './utils';

const testArray = [ { id: 2 }, { id: 3 }, { id: 4 } ];

describe( 'Teaser Posts Grid Block Utility Functions', () => {
	test( 'Update item.', () => {
		const newArray = updateItem( testArray, { id: 1 }, 1 );
		expect( newArray ).toMatchSnapshot()
	} );

	test( 'Update item but don\'t overwrite.', () => {
		const newArray = updateItem( testArray, { foo: 'bar' }, 1 );
		expect( newArray ).toMatchSnapshot()
	} );

	test( 'Append item.', () => {
		const newArray = appendItem( testArray, { id: 6 } );
		expect( newArray ).toMatchSnapshot()
	} );

	test( 'Delete item.', () => {
		const newArray = deleteItem( testArray, 2 );
		expect( newArray ).toMatchSnapshot()
	} );
} );
