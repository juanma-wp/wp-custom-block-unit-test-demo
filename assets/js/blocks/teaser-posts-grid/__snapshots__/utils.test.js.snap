// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teaser Posts Grid Block Utility Functions Append item. 1`] = `
Array [
  Object {
    "id": 2,
  },
  Object {
    "id": 3,
  },
  Object {
    "id": 4,
  },
  Object {
    "id": 6,
  },
]
`;

exports[`Teaser Posts Grid Block Utility Functions Delete item. 1`] = `
Array [
  Object {
    "id": 2,
  },
  Object {
    "id": 3,
  },
]
`;

exports[`Teaser Posts Grid Block Utility Functions Update item but don't overwrite. 1`] = `
Array [
  Object {
    "id": 2,
  },
  Object {
    "foo": "bar",
    "id": 3,
  },
  Object {
    "id": 4,
  },
]
`;

exports[`Teaser Posts Grid Block Utility Functions Update item. 1`] = `
Array [
  Object {
    "id": 2,
  },
  Object {
    "id": 1,
  },
  Object {
    "id": 4,
  },
]
`;
